inc/hclib-place.h

```diff
@@ -48,10 +44,6 @@ typedef struct place_t {
 	int level; /* Level in the HPT tree. Logical root is level 0. */
 	int nChildren;
 	short type;
-#ifdef HC_CUDA
```

Comment:
My goal with this patch was to remove the majority of the dead feature code that's cluttering up the codebase. While these types aren't used in the runtime anymore, they *are* used by the `hwloc_to_hpt` tool. I don't want to start messing with `hpt.dtd` in this commit. If you want to follow up with a commit that fixes all of those loose ends, I'd be fine with that—but I think you want to dump the XML descriptions in favor of your JSON machine description format somewhere down the road anyway, so maybe we can just leave the HPT's XML description as-is until we retire it.